--- Makefile.orig	2013-05-06 15:34:17.000000000 +0000
+++ Makefile
@@ -13,7 +13,6 @@ LICENSE=	GPLv2
 
 OPTIONS_DEFINE=	READLINE
 OPTIONS_DEFAULT=READLINE
-READLINE_DESC=	Enable readline support
 
 USE_BZIP2=	yes
 USES=		gettext ncurses
@@ -40,7 +39,6 @@ PLIST_FILES+=	%%NLS%%share/locale/${lang
 .include <bsd.port.options.mk>
 
 .if ${PORT_OPTIONS:MNLS}
-USES+=		gettext
 PLIST_SUB+=	NLS=""
 .else
 CONFIGURE_ARGS+=	--disable-nls
@@ -49,7 +47,7 @@ PLIST_SUB+=	NLS="@comment "
 
 
 .if ${PORT_OPTIONS:MREADLINE}
-USES=		readline
+USES+=		readline
 .else
 CONFIGURE_ARGS+=--without-readline
 .endif
