--- Makefile.in.orig	2017-01-05 13:19:52.420507000 +0200
+++ Makefile.in
@@ -201,6 +201,7 @@ check_PROGRAMS = test/run-tests$(EXEEXT)
 
 @DARWIN_TRUE@am__append_21 = -lutil
 @DRAGONFLY_TRUE@am__append_22 = include/uv-bsd.h
+@DRAGONFLY_TRUE@am__append_25 = src/unix/freebsd.c src/unix/kqueue.c
 @DRAGONFLY_TRUE@am__append_23 = -lutil
 @FREEBSD_TRUE@am__append_24 = include/uv-bsd.h
 @FREEBSD_TRUE@am__append_25 = src/unix/freebsd.c src/unix/kqueue.c
@@ -344,6 +345,8 @@ am__dirstamp = $(am__leading_dot)dirstam
 @DARWIN_TRUE@	src/unix/libuv_la-fsevents.lo \
 @DARWIN_TRUE@	src/unix/libuv_la-kqueue.lo \
 @DARWIN_TRUE@	src/unix/libuv_la-proctitle.lo
+@DRAGONFLY_TRUE@am__objects_6 = src/unix/libuv_la-freebsd.lo \
+@DRAGONFLY_TRUE@	src/unix/libuv_la-kqueue.lo
 @FREEBSD_TRUE@am__objects_6 = src/unix/libuv_la-freebsd.lo \
 @FREEBSD_TRUE@	src/unix/libuv_la-kqueue.lo
 @LINUX_TRUE@am__objects_7 = src/unix/libuv_la-linux-core.lo \
--- include/uv-bsd.h.orig	2016-08-18 06:38:44.000000000 +0300
+++ include/uv-bsd.h
@@ -23,7 +23,9 @@
 #define UV_BSD_H
 
 #include <pthread_np.h>
+#ifndef __DragonFly__
 typedef cpuset_t cpu_set_t;
+#endif
 
 #define UV_PLATFORM_FS_EVENT_FIELDS                                           \
   uv__io_t event_watcher;                                                     \
--- src/uv-common.c.orig	2016-08-18 06:38:44.000000000 +0300
+++ src/uv-common.c
@@ -371,7 +371,7 @@ int uv_thread_setaffinity(uv_thread_t *t
 
   return 0;
 #else
-#if defined(__APPLE__) || defined(_AIX)
+#if defined(__APPLE__) || defined(_AIX) || defined(__DragonFly__)
   return -ENOTSUP;
 #else
   int i, r;
@@ -420,7 +420,7 @@ int uv_thread_getaffinity(uv_thread_t *t
 
   return 0;
 #else
-#if defined(__APPLE__) || defined(_AIX)
+#if defined(__APPLE__) || defined(_AIX) || defined(__DragonFly__)
   return -ENOTSUP;
 #else
   int i;
--- include/uv.h.orig	2016-08-18 06:38:44.000000000 +0300
+++ include/uv.h
@@ -1468,7 +1468,7 @@ UV_EXTERN void uv_key_set(uv_key_t* key,
 #  define UV_CPU_SETSIZE        32
 # endif
 #else
-# if defined(__APPLE__) || defined(_AIX)
+# if defined(__APPLE__) || defined(_AIX) || defined(__DragonFly__)
 #  define UV_CPU_SETSIZE        8
 # else
 #  define UV_CPU_SETSIZE        CPU_SETSIZE
--- src/unix/freebsd.c	2016-08-18 06:38:44.000000000 +0300
+++ src/unix/freebsd.c
@@ -75,7 +75,8 @@ uint64_t uv__hrtime(uv_clocktype_t type)
 
 
 #ifdef __DragonFly__
-int uv_exepath(char* buffer, size_t* size) {
+/* deprecated long ago, use KERN_PROC_PATHNAME sysctl */
+int uv_exepath_procfs(char* buffer, size_t* size) {
   char abspath[PATH_MAX * 2 + 1];
   ssize_t abspath_size;
 
@@ -97,7 +98,8 @@ int uv_exepath(char* buffer, size_t* siz
 
   return 0;
 }
-#else
+#endif
+
 int uv_exepath(char* buffer, size_t* size) {
   char abspath[PATH_MAX * 2 + 1];
   int mib[4];
@@ -127,7 +129,6 @@ int uv_exepath(char* buffer, size_t* siz
 
   return 0;
 }
-#endif
 
 uint64_t uv_get_free_memory(void) {
   int freecount;
