--- Makefile.sqldrivers.orig	2014-03-15 01:02:55.967990000 +0000
+++ Makefile.sqldrivers
@@ -1,9 +1,6 @@
 # $FreeBSD$
 
-# tell bmake we use the old :L :U modifiers
-.MAKE.FreeBSD_UL= yes
-
-PORTNAME?=	${DB:L}
+PORTNAME?=	${DB:tl}
 DISTVERSION?=	${QT5_VERSION}
 CATEGORIES?=	databases
 PKGNAMEPREFIX?=	qt5-sqldrivers-
@@ -16,12 +13,12 @@ QT_DIST?=	base
 HAS_CONFIGURE=	yes
 CONFIGURE_ARGS+=-no-gui -no-xcb
 
-BUILD_WRKSRC?=	${WRKSRC}/src/plugins/sqldrivers/${DB:L}
+BUILD_WRKSRC?=	${WRKSRC}/src/plugins/sqldrivers/${DB:tl}
 INSTALL_WRKSRC?=${BUILD_WRKSRC}
 
 PLIST_FILES+=	${QT_LIBDIR_REL}/cmake/Qt5Sql/Qt5Sql_Q${DB}DriverPlugin.cmake \
-		${QT_PLUGINDIR_REL}/sqldrivers/libqsql${DB:L:C/^sql//}.so \
-		%%DEBUG%%${QT_PLUGINDIR_REL}/sqldrivers/libqsql${DB:L:C/^sql//}.so.debug
+		${QT_PLUGINDIR_REL}/sqldrivers/libqsql${DB:tl:C/^sql//}.so \
+		%%DEBUG%%${QT_PLUGINDIR_REL}/sqldrivers/libqsql${DB:tl:C/^sql//}.so.debug
 PLIST_DIRSTRY+=	${QT_PLUGINDIR_REL}/sqldrivers \
 		${QT_PLUGINDIR_REL} \
 		${QT_ARCHDIR_REL} \
