From 87a769fc0a9cdc75f2f79f5cc8072efa95ff4437 Mon Sep 17 00:00:00 2001
From: Andreas Schneider <asn@samba.org>
Date: Fri, 6 Aug 2021 12:03:49 +0200
Subject: [PATCH] CVE-2020-25719 s4:kdc: Remove trailing spaces in pac-glue.c

BUG: https://bugzilla.samba.org/show_bug.cgi?id=14561

Signed-off-by: Andreas Schneider <asn@samba.org>
Reviewed-by: Andrew Bartlett <abartlet@samba.org>
---
 source4/kdc/pac-glue.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/source4/kdc/pac-glue.c b/source4/kdc/pac-glue.c
index 688103d8477b..4066389e7172 100644
--- a/source4/kdc/pac-glue.c
+++ b/source4/kdc/pac-glue.c
@@ -575,12 +575,12 @@ int samba_krbtgt_is_in_db(struct samba_kdc_entry *p,
 	if (!mem_ctx) {
 		return ENOMEM;
 	}
-	
+
 	trust_direction = ldb_msg_find_attr_as_int(p->msg, "trustDirection", 0);
 
 	if (trust_direction != 0) {
 		/* Domain trust - we cannot check the sig, but we trust it for a correct PAC
-		   
+
 		   This is exactly where we should flag for SID
 		   validation when we do inter-foreest trusts
 		 */
@@ -768,7 +768,7 @@ NTSTATUS samba_kdc_update_pac_blob(TALLOC_CTX *mem_ctx,
 		return nt_status;
 	}
 
-	nt_status = samba_get_logon_info_pac_blob(mem_ctx, 
+	nt_status = samba_get_logon_info_pac_blob(mem_ctx,
 						  user_info_dc, pac_blob);
 
 	return nt_status;
