--- auto/openssl.orig	2022-10-03 21:42:59.518007000 +0200
+++ auto/openssl	2022-10-03 21:43:17.437601000 +0200
@@ -12,8 +12,8 @@
     njs_feature="OpenSSL library"
     njs_feature_name=NJS_HAVE_OPENSSL
     njs_feature_run=yes
-    njs_feature_incs=
-    njs_feature_libs=""
+    njs_feature_incs="-I/usr/local/include"
+    njs_feature_libs="-L/usr/local/lib -lcrypto"
     njs_feature_test="#include <openssl/evp.h>
 
                       int main() {
