--- skia/src/sksl/dsl/DSLExpression.cpp.orig	2021-11-25 15:00:28 UTC
+++ skia/src/sksl/dsl/DSLExpression.cpp
@@ -45,10 +45,10 @@ DSLExpression::DSLExpression(float value
     : fExpression(SkSL::Literal::MakeFloat(ThreadContext::Context(),
                                            pos.line(),
                                            value)) {
-    if (!isfinite(value)) {
-        if (isinf(value)) {
+    if (!std::isfinite(value)) {
+	if (std::isinf(value)) {
             ThreadContext::ReportError("floating point value is infinite");
-        } else if (isnan(value)) {
+        } else if (std::isnan(value)) {
             ThreadContext::ReportError("floating point value is NaN");
         }
     }
